app/Jobs/RefreshNftMetadata.php

```diff
@@ -40,40 +43,48 @@ public function __construct(
      */
     public function handle(AlchemyWeb3DataProvider $provider): void
     {
-
         if (SpamContract::isSpam($this->collection->address, $this->collection->network)) {
-            Log::info('RefreshNftMetadata Job: Ignored for spam contract', [
-                'address' => $this->collection->address,
-                'network' => $this->collection->network->id,
-                'token_number' => $this->nft->token_number,
-            ]);
+            return;
+        }
+
+        $nfts = Nft::whereNotNull('metadata_requested_at')
```

Comment:
As we discussed in the Slack: 
We need to get NFTs grouped by network as we need to request metadata for NFTs on the same network